labgrid/driver/mqtt.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import queue
+import logging
+import threading
+import time
+
+import attr
+import paho.mqtt.publish as publish
+import paho.mqtt.client as mqtt
+
+from .common import Driver
+from ..factory import target_factory
+from ..protocol import PowerProtocol
+from ..step import step
+from ..util import Timeout
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class TasmotaPowerDriver(Driver, PowerProtocol):
+    bindings = {
+            "power": {"TasmotaPowerPort", "NetworkTasmotaPowerPort"}
+    }
+    delay = attr.ib(default=2.0, validator=attr.validators.instance_of(float))
+    _client = attr.ib(default=mqtt.Client(), validator=attr.validators.instance_of(mqtt.Client))
+    _status_lock = attr.ib(default=threading.Lock())
+    _status = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+
+    def on_activate(self):
+        self._client.on_message = self._on_message
+        self._client.on_connect = self._on_connect
+        self._client.connect(self.power.host)
+        self._client.loop_start()
+
+    def on_deactivate(self):
+        self._client.loop_stop()
+
+    def _on_message(self, client, userdata, msg):
+        if msg.payload == b'ON':
+            status = True
+        elif msg.payload == b'OFF':
+            status = False
+        with self._status_lock:
+            self._status = status
+
+    def _on_connect(self, client, userdata, flags, rc):
+        client.subscribe(self.power.status_topic)
+
+    @Driver.check_active
+    @step()
+    def on(self):
+        self._client.publish(self.power.power_topic, payload="ON")
```

Comment:
Do we get feedback if this worked? When does this return?

Reply:
Yes, we should handle the returned class and wait for the topic to be delivered.